File: example/src/main/java/org/wordpress/android/fluxc/example/AppComponent.java
package org.wordpress.android.fluxc.example;

import org.wordpress.android.fluxc.module.AppContextModule;
import org.wordpress.android.fluxc.module.ReleaseBaseModule;
import org.wordpress.android.fluxc.module.ReleaseNetworkModule;
import org.wordpress.android.fluxc.module.ReleaseOkHttpClientModule;

import javax.inject.Singleton;

import dagger.Component;

@Singleton
@Component(modules = {
        AppContextModule.class,
        AppSecretsModule.class,
        ReleaseOkHttpClientModule.class,
        ReleaseBaseModule.class,
        ReleaseNetworkModule.class
})
public interface AppComponent {
    void inject(ExampleApp object);
    void inject(MainExampleActivity object);
    void inject(SitesFragment object);
    void inject(MainFragment object);
    void inject(MediaFragment object);
    void inject(CommentsFragment object);
    void inject(PostsFragment object);
    void inject(AccountFragment object);
    void inject(SignedOutActionsFragment object);
    void inject(TaxonomiesFragment object);
    void inject(UploadsFragment object);
}

File: example/src/main/java/org/wordpress/android/fluxc/example/AppComponent.kt
package org.wordpress.android.fluxc.example

import dagger.Component
import org.wordpress.android.fluxc.module.AppContextModule
import org.wordpress.android.fluxc.module.ReleaseBaseModule
import org.wordpress.android.fluxc.module.ReleaseNetworkModule
import org.wordpress.android.fluxc.module.ReleaseOkHttpClientModule
import javax.inject.Singleton

@Singleton
@Component(modules = arrayOf(
        AppContextModule::class,
        AppSecretsModule::class,
        ReleaseOkHttpClientModule::class,
        ReleaseBaseModule::class,
        ReleaseNetworkModule::class))
interface AppComponent {
    fun inject(app: ExampleApp)
    fun inject(activity: MainExampleActivity)
    fun inject(fragment: SitesFragment)
    fun inject(fragment: MainFragment)
    fun inject(fragment: MediaFragment)
    fun inject(fragment: CommentsFragment)
    fun inject(fragment: PostsFragment)
    fun inject(fragment: AccountFragment)
    fun inject(fragment: SignedOutActionsFragment)
    fun inject(fragment: TaxonomiesFragment)
    fun inject(fragment: UploadsFragment)
    fun inject(fragment: ThemeFragment)
}
